### PR TITLE
Add AnalyticsManager.

### PR DESCRIPTION
**Pull request by andrew-tran-infillion**
Add AnalyticsManager.

  Add background location permission to Android Sample App.
  Make Android AnalyticsManager spec for old and new arch.
  Make Android AnalyticsManagerModule and add it to GimbalPackage modules.
  Make iOS AnalyticsManager.
  Add APS entitlement to iOS Sample App.
  Add AnalyticsManager test code to Sample App.
  Make NativeAnalyticsManager.ts and add it to src/index.tsx.

### CODE REVIEW

File: sample/android/app/src/debug/AndroidManifest.xml [hunk 1]
```diff
@@ -3,6 +3,7 @@
     xmlns:tools="http://schemas.android.com/tools">
 
     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
+    <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />
```
Comment (kimvsparrow) [inline sample/android/app/src/debug/AndroidManifest.xml]: I'm pretty sure this isn't needed. This permission is already in the main manifest.  When building a debug variant, this debug manifest is applied in addition to the main manifest.

File: sample/ios/RtnGimbalSdkSample/RtnGimbalSdkSample.entitlements [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kimvsparrow) [inline sample/ios/RtnGimbalSdkSample/RtnGimbalSdkSample.entitlements]: Does this do anything yet, considering that we haven't implemented `CommunicationManager`?

Comment (andrew-tran-infillion) [reply sample/ios/RtnGimbalSdkSample/RtnGimbalSdkSample.entitlements]: oops, this was a holdover from when Nick was asking about testing communicates, I'll remove it now and add it back in later, sounded like we're going to have more discussions surrounding CommunicationManager.

File: sample/ios/RtnGimbalSdkSample.xcworkspace/contents.xcworkspacedata [hunk 3]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Workspace
    version = "1.0">
+   <FileRef
+      location = "group:../../../../../../Documents/firestation.gpx">
```
Comment (kimvsparrow) [inline sample/ios/RtnGimbalSdkSample.xcworkspace/contents.xcworkspacedata]: This looks like a personal testing file, if so it can be removed.

**Review by kimvsparrow — APPROVED** [review]
Looks good, thanks Andrew.

**Pull request merged**